hibernate-core/src/main/java/org/hibernate/persister/entity/JoinedSubclassEntityPersister.java

```diff
@@ -1355,15 +1358,22 @@ public void pruneForSubclasses(TableGroup tableGroup, Map<String, EntityNameUse>
 					}
 				}
 			}
+			final String tableName = persister.getTableName();
 			final TableReference mainTableReference = tableGroup.getTableReference(
 					null,
-					persister.getTableName(),
+					tableName,
 					false
 			);
-			if ( mainTableReference == null ) {
-				throw new UnknownTableReferenceException( persister.getTableName(), "Couldn't find table reference" );
```

Comment:
I guess you removed the exception throwing because of not doing `resolveTableReference` anymore for `EntityNameUse.TREAT`?